Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp

```diff
@@ -1321,5 +1323,17 @@ void NetworkStorageManager::cacheStorageRepresentation(CompletionHandler<void(St
     callback(builder.toString());
 }
 
+#if ENABLE(SERVICE_WORKER)
+void NetworkStorageManager::dispatchTaskToBackgroundFetchManager(const WebCore::ClientOrigin& origin, Function<void(BackgroundFetchStoreManager&)>&& callback)
```

Comment:
`NetworkStorageManager` should not accept new task on its queue when `m_closed` is true (it dispatches a cleanup task to the queue when close, which is supposed to be the last task).
I would prefer adding individual functions like `NetworkStorageManager::backgroundFetchClearFetch` to not expose `BackgroundFetchManager`  to other class (as technically they will be able to capture it). 

We may do that as followup and return `BackgroundFetchStoreManager*` here.

Reply:
I'll add a `if(m_closed)` check.
I'll also change BackgroundFetchStoreManager to not be ref counted.